src/utils.js

```diff
@@ -15,3 +20,68 @@ module.exports = function goToPath(path, obj) {
   }
   return current;
 };
+
+
+/**
+ * N.B. baseGetTag(), isObject() & isFunction() are lifted straight out of Lodash,
+ * with a little tweaking to conform to the coding standards in this project
+ *
+ * Lodash is awesome but I did not want add an extra dependency to this project!
+ */
+
+/**
+ * The base implementation of `getTag` without fallbacks for buggy environments.
+ *
+ * @private
+ * @param {*} value The value to query.
+ * @returns {string} Returns the `toStringTag`.
+ */
+function baseGetTag(value) {
+  if (value == null) {
+    return value === undefined ? '[object Undefined]' : '[object Null]'
+  }
+  if (!(symToStringTag && symToStringTag in Object(value))) {
+    return toString.call(value)
+  }
+  var isOwn = hasOwnProperty.call(value, symToStringTag)
+  var tag = value[symToStringTag]
+  var unmasked = false
+
+  try {
+    value[symToStringTag] = undefined
+    unmasked = true
+  } catch (e) {}
+
+  const result = toString.call(value)
+  if (unmasked) {
+    if (isOwn) {
+      value[symToStringTag] = tag
+    } else {
+      delete value[symToStringTag]
+    }
+  }
+  return result
+}
+
+function isObject(value) {
+  var type = typeof value;
+
+  return value != null && (type == 'object' || type == 'function');
```

Comment:
lodash code again - the check makes sense (besides the `lodash` code has been battle-tested)